### ISSUE TITLE
Missing erlang header files

### ISSUE THREAD

**Issue by kfatehi** [nerves-project/nerves-toolchain#3]
This is the output on a Mac when trying to compile a project that uses elixir_ale

```
✘ 21:53 ~/Workspace/codelock [master|✔] $ make
mix deps.get
Running dependency resolution
All dependencies up to date
mix compile
==> elixir_ale
Makefile:17: *** Could not find include directory for ei.h. Check that Erlang header files are available.  Stop.
could not compile dependency :elixir_ale, "mix compile" failed. You can recompile this dependency with "mix deps.compile elixir_ale", update it with "mix deps.update elixir_ale" or clean it with "mix deps.clean elixir_ale"
** (MatchError) no match of right hand side value: 2
    mix.exs:4: Mix.Tasks.Compile.ElixirAle.run/1
    (elixir) lib/enum.ex:1043: anonymous fn/3 in Enum.map/2
    (elixir) lib/enum.ex:1387: Enum."-reduce/3-lists^foldl/2-0-"/3
    (elixir) lib/enum.ex:1043: Enum.map/2
    (mix) lib/mix/tasks/compile.all.ex:19: anonymous fn/1 in Mix.Tasks.Compile.All.run/1
    (mix) lib/mix/tasks/compile.all.ex:37: Mix.Tasks.Compile.All.with_logger_app/1
    (mix) lib/mix/tasks/compile.ex:64: Mix.Tasks.Compile.run/1
    (mix) lib/mix/tasks/deps.compile.ex:102: anonymous fn/2 in Mix.Tasks.Deps.Compile.do_mix/2

make: *** [compile] Error 1
```

**Comment by nickgal** [comment]
I was able to get past this by setting a couple environment variables in my project's `Makefile`

```shell
ifeq ($(NERVES_ROOT),)
    $(error Make sure that you source nerves-env.sh first)
endif

# https://github.com/nerves-project/nerves-sdk/blob/5b59efa409d5374c8adb194d8f2f733972b7e38f/nerves.mk#L19
ERL_INTERFACE_DIR=$(wildcard $(NERVES_SDK_SYSROOT)/usr/lib/erlang/lib/erl_interface-*)

# https://github.com/fhunleth/elixir_ale/blob/fae035b82f9a22c8af674adb9847fd7516c98e35/Makefile#L13-L14
ERL_EI_INCLUDE_DIR=$(ERL_INTERFACE_DIR)/include
ERL_EI_LIBDIR=$(ERL_INTERFACE_DIR)/lib

export ERL_EI_INCLUDE_DIR
export ERL_EI_LIBDIR

include $(NERVES_ROOT)/scripts/nerves-elixir.mk
```

**Comment by kfatehi** [comment]
Worked!! Thanks @nickgal

**Comment by fhunleth** [comment]
@nickgal: Thanks for posting the environment variable that I unintentionally missed. I added it to the list of variables exported by nerves-env.sh. Fix is nerves-project/nerves-sdk@80fefb08bbcd932767bf9c4f36ffc692fdbd0613.
